core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -54,7 +54,8 @@ function makeIdNumeric(id) {
 }
 
 function formatDate(date) {
-  return date.split(".")[0].replace("T", " ");
+  if (date <= 0) return "-";
```

Comment:
How about using "n/a" which would specifically indicate that its invalid to expect the end time ?